pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package domainmapping
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	network "knative.dev/networking/pkg"
+	netv1alpha1 "knative.dev/networking/pkg/apis/networking/v1alpha1"
+	netclientset "knative.dev/networking/pkg/client/clientset/versioned"
+	networkinglisters "knative.dev/networking/pkg/client/listers/networking/v1alpha1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/reconciler"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+	domainmappingreconciler "knative.dev/serving/pkg/client/injection/reconciler/serving/v1alpha1/domainmapping"
+	"knative.dev/serving/pkg/reconciler/domainmapping/resources"
+)
+
+// Reconciler implements controller.Reconciler for DomainMapping resources.
+type Reconciler struct {
+	ingressLister networkinglisters.IngressLister
+	netclient     netclientset.Interface
+}
+
+// Check that our Reconciler implements Interface
+var _ domainmappingreconciler.Interface = (*Reconciler)(nil)
+
+// ReconcileKind implements Interface.ReconcileKind.
+func (r *Reconciler) ReconcileKind(ctx context.Context, dm *v1alpha1.DomainMapping) reconciler.Event {
+	logger := logging.FromContext(ctx)
+	logger.Debugf("Reconciling DomainMapping %s/%s", dm.Namespace, dm.Name)
+
+	// Mapped URL is the metadata.name of the DomainMapping.
+	url := &apis.URL{
+		Scheme: "http",
+		Host:   dm.Name,
+	}
+
+	dm.Status.URL = url
+	dm.Status.Address = &duckv1.Addressable{URL: url}
+
+	// TODO(jz) allow overriding via annotation, configmap.
+	ingressClass := network.IstioIngressClassName
+
+	logger.Debugf("Mapping %s to %s/%s", url, dm.Spec.Ref.Namespace, dm.Spec.Ref.Name)
+	desired := resources.MakeIngress(dm, ingressClass)
+	_, err := r.reconcileIngress(ctx, dm, desired)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
```suggestion
	_, err := r.reconcileIngress(ctx, dm, desired)
	return err
```

Reply:
This is temporary till I propagate ingress status in next PR

Reply:
.. but meh, might as well change it till then - done